### PR TITLE
[core] SetHandle returns DEVICE_FAILED on unsupported systems

### PR DESCRIPTION
**Pull request by dmitryermilov**
Fixes: #2230

Signed-off-by: Dmitry Ermilov <dmitry.ermilov@intel.com>

### CODE REVIEW

**Comment by dmitryermilov** [comment]
@varistar , FYI. We might need to same for VPL spec.

File: doc/mediasdk-man.md [hunk 1]
```diff
@@ -2050,6 +2050,7 @@ If the specified system handle is a COM interface, the reference counter of the
 --- | ---
 `MFX_ERR_NONE` | The function completed successfully.
 `MFX_ERR_UNDEFINED_BEHAVIOR` | The same handle is redefined. For example, the function has been called twice with the same handle type or internal handle has been created by the SDK before this function call.
+`MFX_ERR_DEVICE_FAILED` | The SDK cannot initialize using the handle.
```
Comment (dvrogozh) [inline doc/mediasdk-man.md:2053]: Generally speaking this is a change to the msdk behavior. I personally Ok with the change, but we need wider review and approval. @varistar, @onabiull : your opinion?

**Review by dvrogozh — APPROVED** [review]

File: doc/mediasdk-man.md (hunk 1 above)
Comment (varistar) [reply doc/mediasdk-man.md:2053]: I'm fine with it, it looks logical. I'll update oneVPL spec accardingly

Comment (dvrogozh) [reply doc/mediasdk-man.md:2053]: @varistar : do you approve this PR (you did not add a mark)? Or we should wait for spec update?

**Pull request merged**